agents/random.py

```diff
@@ -20,13 +20,25 @@ def __init__(
     def forward(self, inputs: Dict) -> Dict:
         neighbors = inputs[self._neighbors_prefix]  # (batch_size, neighbors_cnt)
 
-        neighbors_mask = inputs[f'{self._neighbors_prefix}.mask']  # (batch_size, neighbors_cnt)
-        neighbors_mask_sum = torch.sum(neighbors_mask, dim=-1)  # (batch_size)
+        neighbors_length = neighbors.lengths
 
-        next_neighbors = torch.from_numpy(np.random.randint(
-            low=np.zeros_like(neighbors_mask_sum.cpu().detach().numpy()),
-            high=neighbors_mask_sum.cpu().detach().numpy()
-        )).to(neighbors.device)  # (batch_size)
+        low = torch.zeros_like(neighbors_length)
+        high = neighbors_length
+        next_neighbors = torch.randint(low=low.item(), high=high.item(), size=(1,))
+
+        next_neighbor = torch.gather(neighbors.flatten_values, 1, torch.unsqueeze(next_neighbors, 0))
```

Comment:
Let's use keywords for arguments in this function

```suggestion
        next_neighbors = torch.squeeze(
            torch.gather(
                inputs=neighbors,  # (batch_size, num_neighbors)
                dim=1, 
                index=torch.unsqueeze(next_neighbors, dim=1)  # (batch_size, 1)
            )  # (batch_size, 1)
        )  # (batch_size)
```